s3prl/downstream/mos_prediction/expert.py

```diff
@@ -0,0 +1,343 @@
+import warnings
+from collections import defaultdict
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+from scipy.stats import pearsonr, spearmanr
+import torch
+import torch.nn as nn
+from torch.distributed import is_initialized
+from torch.nn.utils.rnn import pad_sequence
+from torch.utils.data import DataLoader, DistributedSampler
+from tqdm import tqdm
+
+from .dataset import VCC18SegmentalDataset, VCC16SegmentalDataset
+from .model import Model
+
+warnings.filterwarnings("ignore")
+
+
+class DownstreamExpert(nn.Module):
+    def __init__(self, upstream_dim, downstream_expert, **kwargs):
+        super(DownstreamExpert, self).__init__()
+        self.upstream_dim = upstream_dim
+        self.datarc = downstream_expert["datarc"]
+        self.modelrc = downstream_expert["modelrc"]
+        print(self.datarc['save_dir'])
+        idtable = Path(self.datarc['save_dir']) / 'idtable.pkl'
```

Comment:
Actually the DownstreamExpert also gets the `expdir` argument from command line. You can just catch that at the __init__ function, and won't need to bother syncing the `expdir` and `save_dir` on the command line which cause complication.

Reply:
Also, I guess this `idtable` is very small right? Will you consider to directly add it into git to ensure future reproducibility?